wavepool/views.py

```diff
@@ -14,9 +14,20 @@ def front_page(request):
             archive: the rest of the newsposts, sorted by most recent
     """
     template = loader.get_template('wavepool/frontpage.html')
-    cover_story = NewsPost.objects.all().order_by('?').first()
-    top_stories = NewsPost.objects.all().order_by('?')[:3]
-    other_stories = NewsPost.objects.all().order_by('?')
+    cover_story = None
+    top_stories = []
+    other_stories = []
+    newsposts = NewsPost.objects.all().order_by('publish_date')
+
+    i = 1
+    for n in newsposts:
+        if n.is_cover_story == True:
+            cover_story = n
+        if i < 3:
+            top_stories.append(n)
+            i = i + 1
+        else:
+            other_stories.append(n)
```

Comment:
This code could be cleaned up a little bit with a little bit of django magic. To save on database hits, you could cast the whole query set to a list first, and then split it out. e.g.
```
newsposts =  NewsPost.objects.exclude(is_cover_story=True).order_by('-publish_date')
top_stories = newsposts[:3]
other_stories = newsposts[3:]
```